From ee0d0b12a2d5b411f4dc0928d62a433a62e5ae79 Mon Sep 17 00:00:00 2001
From: Alex Leibovich <alexl@marvell.com>
Date: Tue, 2 Apr 2019 10:58:46 +0300
Subject: [PATCH 134/386] arm64: dts: cn9130: add cn9130-db device tree with
 setup B

cn913x DB board is a modular board,
which supports multiple possible configurations.

topology for setup B:
CP0-SerDes:
	- lane0: PCIe0
	- lane1: PCIe0
	- lane2: PCIe0
	- lane3: PCIe0
	- lane4: SFI0
	- lane5: SATA1
AP-MPPs:
	- SDIO, UART
CP0-MPPs:
	- RGMII1, SDIO, I2C, NAND

Change-Id: Ia5eed9e41e607af9dd526128ad272b6ca92e6e30
Signed-off-by: Alex Leibovich <alexl@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/6738
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 .../bindings/arm/marvell/cn913x.txt           |   5 +
 arch/arm64/boot/dts/marvell/Makefile          |   1 +
 arch/arm64/boot/dts/marvell/cn9130-db-B.dts   | 154 ++++++++++++++++++
 arch/arm64/boot/dts/marvell/cn9130.dtsi       |   9 +-
 4 files changed, 167 insertions(+), 2 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/cn9130-db-B.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/cn913x.txt b/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
index 3a34091762bf..15e9f2506cd7 100644
--- a/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/cn913x.txt
@@ -15,6 +15,11 @@ Boards:
       when the board being used is the OcteonTX2 CN9130 Development board with
       Setup A.
 
+Boards:
+   - "marvell,cn9130-db-B"
+      when the board being used is the OcteonTX2 CN9130 Development board with
+      Setup B.
+
    - "marvell,cn9131-db-A"
       when the board being used is the OcteonTX2 CN9131 Development board with
       Setup A.
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 568bfdcbae35..657f3b314e4e 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -25,5 +25,6 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-ucpe.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-vd-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-3900-db-vd-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-A.dtb
+dtb-$(CONFIG_ARCH_MVEBU) += cn9130-db-B.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9131-db-A.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += cn9132-db-A.dtb
diff --git a/arch/arm64/boot/dts/marvell/cn9130-db-B.dts b/arch/arm64/boot/dts/marvell/cn9130-db-B.dts
new file mode 100644
index 000000000000..f6b7b3c99624
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/cn9130-db-B.dts
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2019 Marvell International Ltd.
+ *
+ * SPDX-License-Identifier:    GPL-2.0
+ * https://spdx.org/licenses
+ */
+
+#include "cn9130-db.dtsi"
+
+/ {
+	model = "Model: Marvell CN9130 development board (CP NOR) setup(B)";
+	compatible = "marvell,cn9130-db-B";
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	aliases {
+		i2c0 = &cp0_i2c0;
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp0_eth1;
+		ethernet2 = &cp0_eth2;
+	};
+
+	memory@00000000 {
+		device_type = "memory";
+		reg = <0x0 0x0 0x0 0x80000000>;
+	};
+};
+
+&uart0 {
+	status = "okay";
+};
+
+/* on-board eMMC - U9 */
+&ap_sdhci0 {
+	pinctrl-names = "default";
+	bus-width = <8>;
+	status = "okay";
+};
+
+/*
+ * CP related configuration
+ */
+&cp0_i2c0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&cp0_i2c0_pins>;
+	status = "okay";
+	clock-frequency = <100000>;
+};
+
+&cp0_i2c1 {
+	status = "okay";
+};
+
+/* CON 28 */
+&cp0_sdhci0 {
+	status = "okay";
+};
+/* U54 */
+&cp0_nand_controller {
+	pinctrl-names = "default";
+	pinctrl-0 = <&nand_pins &nand_rb>;
+	status = "okay";
+
+	nand@0 {
+		reg = <0>;
+		label = "main-storage";
+		nand-rb = <0>;
+		nand-ecc-mode = "hw";
+		nand-on-flash-bbt;
+		nand-ecc-strength = <8>;
+		nand-ecc-step-size = <512>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "U-Boot";
+				reg = <0 0x200000>;
+			};
+			partition@200000 {
+				label = "Linux";
+				reg = <0x200000 0xd00000>;
+			};
+			partition@1000000 {
+				label = "Filesystem";
+				reg = <0x1000000 0x3f000000>;
+			};
+		};
+	};
+};
+
+/* SLM-1521-V2, CON6 */
+&cp0_pcie0 {
+	status = "okay";
+	num-lanes = <4>;
+	num-viewport = <8>;
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy0 0
+		&cp0_comphy1 0
+		&cp0_comphy2 0
+		&cp0_comphy3 0>;
+};
+
+&cp0_sata0 {
+	status = "okay";
+	/* SLM-1521-V2, CON2 */
+	sata-port@1 {
+		status = "okay";
+		/* Generic PHY, providing serdes lanes */
+		phys = <&cp0_comphy5 1>;
+	};
+};
+
+&cp0_mdio {
+	status = "okay";
+	phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+	phy1: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
+&cp0_ethernet {
+	status = "okay";
+};
+
+/* SLM-1521-V2, CON9 */
+&cp0_eth0 {
+	status = "okay";
+	phy-mode = "10gbase-kr";
+	/* Generic PHY, providing serdes lanes */
+	phys = <&cp0_comphy4 0>;
+	managed = "in-band-status";
+	sfp = <&cp0_sfp_eth0>;
+};
+
+/* CON56 */
+&cp0_eth1 {
+	status = "okay";
+	phy = <&phy0>;
+	phy-mode = "rgmii-id";
+};
+
+/* CON57 */
+&cp0_eth2 {
+	status = "okay";
+	phy = <&phy1>;
+	phy-mode = "rgmii-id";
+};
diff --git a/arch/arm64/boot/dts/marvell/cn9130.dtsi b/arch/arm64/boot/dts/marvell/cn9130.dtsi
index 25fed7b291d2..44826a0a376c 100644
--- a/arch/arm64/boot/dts/marvell/cn9130.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9130.dtsi
@@ -101,10 +101,15 @@
 		};
 		nand_pins: nand-pins {
 			marvell,pins =
-			"mpp17", "mpp18", "mpp19", "mpp20", "mpp21", "mpp22",
-			"mpp23", "mpp24", "mpp25", "mpp26", "mpp27";
+			"mpp15", "mpp16", "mpp17", "mpp18", "mpp19",
+			"mpp20", "mpp21", "mpp22", "mpp23", "mpp24",
+			"mpp25", "mpp26", "mpp27";
 			marvell,function = "dev";
 		};
+		nand_rb: nand-rb {
+			marvell,pins = "mpp13";
+			marvell,function = "nf";
+		};
 	};
 };
 
-- 
2.17.1

